src/System.Private.CoreLib/shared/System/Security/PermissionSet.cs

```diff
@@ -18,38 +19,38 @@ public partial class PermissionSet : ICollection, IEnumerable, IDeserializationC
         public virtual int Count { get { return 0; } }
         public virtual bool IsReadOnly { get { return false; } }
         public virtual bool IsSynchronized { get { return false; } }
-        public virtual object SyncRoot { get { return null; } }
-        public IPermission AddPermission(IPermission perm) { return AddPermissionImpl(perm); }
-        protected virtual IPermission AddPermissionImpl(IPermission perm) { return default(IPermission); }
+        public virtual object? SyncRoot { get { return null; } }
+        public IPermission? AddPermission(IPermission perm) { return AddPermissionImpl(perm); }
+        protected virtual IPermission? AddPermissionImpl(IPermission perm) { return default(IPermission); }
         public void Assert() { }
         public bool ContainsNonCodeAccessPermissions() { return false; }
         [Obsolete]
-        public static byte[] ConvertPermissionSet(string inFormat, byte[] inData, string outFormat) { return null; }
-        public virtual PermissionSet Copy() { return default(PermissionSet); }
+        public static byte[]? ConvertPermissionSet(string inFormat, byte[] inData, string outFormat) { return null; }
+        public virtual PermissionSet? Copy() { return default(PermissionSet); }
```

Comment:
The implementation here should be `return new PermissionSet(this);` and then it shouldn't be nullable.

Reply:
addressed separately in: https://github.com/dotnet/coreclr/pull/23637 (for workarounded to mark as non-nullable)